lib/backend/compat/compat.go

```diff
@@ -199,6 +239,33 @@ func (c *ModelAdaptor) getProfile(k model.Key) (*model.KVPair, error) {
 	return &d, nil
 }
 
+// getBlockAffinity gets KVPair for Block. It gets the block value first,
+// then checks for `Affinity` field first, then `HostAffinity` as a backup.
+// For more details see: https://github.com/projectcalico/libcalico-go/issues/226
+func (c *ModelAdaptor) getBlockAffinity(k model.Key) (*model.KVPair, error) {
+	bk := k.(model.BlockKey)
+
+	v, err := c.client.Get(model.BlockKey{CIDR: bk.CIDR})
+	if err != nil {
+		return nil, err
+	}
+	val := v.Value.(*model.AllocationBlock)
+
+	// Check for `Affinity` field first (this is to make sure we're
+	// compatible with Python version etcd data-model).
+	if val.Affinity == nil {
+
+		// Check for hostAffinity if Affinity is nil.
+		if val.HostAffinity != nil {
+			// Convert HostAffinity=hostname into Affinity=host:hostname format.
+			hostAffinityStr := affinedTo + ":" + *val.HostAffinity
```

Comment:
It's especially weird because we've got two constants in two different places defined to the same thing.  I think it is more clear for people reading the code to see what value is actually being put there, rather than having to go look at that variable to see what it is.